assembl/static2/js/app/components/form/utils.js

```diff
@@ -71,11 +75,40 @@ export function convertToEntries(valuesByLocale: I18nValue): LangstringEntries {
   }));
 }
 
-export function convertRichTextToEntries(valuesByLocale: I18nRichTextValue): LangstringEntries {
-  return Object.keys(valuesByLocale).map(locale => ({
-    localeCode: locale,
-    value: convertEditorStateToHTML(valuesByLocale[locale])
+type RichTextVariables = {
+  attachments: Array<string>,
+  entries: LangstringEntries
+};
+
+/*
+  Upload new attachments and return variables for mutation
+*/
+export async function convertRichTextToVariables(
+  valuesByLocale: I18nRichTextValue,
+  client: ApolloClient
+): Promise<RichTextVariables> {
+  const uploadDocument = options => client.mutate({ mutation: UploadDocument, ...options });
+  const results = Object.keys(valuesByLocale).map(async (locale) => {
+    const result = await uploadNewAttachments(valuesByLocale[locale], uploadDocument);
+    const { documentIds, contentState } = result;
+    return {
+      documentIds: documentIds,
+      localeCode: locale,
+      value: convertEditorStateToHTML(EditorState.createWithContent(contentState))
+    };
+  });
+
+  const variables = await Promise.all(results);
```

Comment:
I hope we don't have db conflicts when uploading multiple attachments with uwsgi.

Reply:
just tested with multiple attachments in multiple fields in different languages and it seems ok but it's `pserve` and not `uwsgi`...

Reply:
pserve is using threads, uwsgi is used processes. Be sure to test it once deployed.